src/layoutparser/ocr/paddleocr_agent.py

```diff
@@ -0,0 +1,121 @@
+        """Create a Tesseract OCR Agent.
+
+        Args:
+            languages (:obj:`list` or :obj:`str`, optional):
+                You can specify the language code(s) of the documents to detect to improve
+                accuracy. The supported language and their code can be found on
+                `its github repo <https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.1/doc/doc_ch/whl.md>`_.
+                It supports llaguages：`ch`, `en`, `french`, `german`, `korean`, `japan`.
+                Defaults to 'eng'.
+        """
+        self.lang = languages
+        self.use_gpu = use_gpu
+        self.use_angle_cls = use_angle_cls
+        self.configs = kwargs
+        self.ocr = paddleocr.PaddleOCR(use_gpu=self.use_gpu, use_angle_cls=self.use_angle_cls, lang=self.lang)
+   
+    def resized_long(self, image, target=480):
+        shape = image.shape
+        if max(image.shape[0], image.shape[1]) >= target:
+            return image
+        if shape[0] >= shape[1]:
+            ratio = 1.0 * target / shape[0]
+            out = [int(shape[1] * ratio), target]
+        else:
+            ratio = 1.0 * target / shape[1]
+            out = [target, int(shape[0] * ratio)]
+        return cv2.resize(image, out)
+
+    def pad_img_to_longer_edge(self, image):
+        max_shape = max(image.shape[0], image.shape[1])
+        out_img = np.ones([max_shape, max_shape, 3]) * 127
+        out_img[:image.shape[0], :image.shape[1], :image.shape[2]] = image
+        return out_img
+
+    def detect(
+        self, image, det=True, rec=True, cls=True,
+        return_response=False, return_only_text=True
+    ):
+        """Send the input image for OCR.
+
+        Args:
+            image (:obj:`np.ndarray` or :obj:`str`):
+                The input image array or the name of the image file
+            det (:obj:`bool`, optional): 
+                use text detection or not, if false, only rec will be exec.
+                Default to `True`.
+            rec (:obj:`bool`, optional): 
+                Use text recognition or not, if false, only det will be exec. 
+                Default to `True`.
+            cls (:obj:`bool`, optional):
+                Use 180 degree rotation text recognition or not. 
+                Default to `True`.
+            return_response (:obj:`bool`, optional):
+                Whether directly return all output (string and boxes
+                info) from Tesseract.
+                Defaults to `False`.
+            return_only_text (:obj:`bool`, optional):
+                Whether return only the texts in the OCR results.
+                Defaults to `False`.
+        """ 
+        image = self.resized_long(image)
+        image = self.pad_img_to_longer_edge(image)
+        res = self.ocr.ocr(image, det=det, rec=rec, cls=cls)
+
+        if return_response:
+            return res
+        
+        if return_only_text:
```

Comment:
1. A score is needed to filter the recognition results and corresponded polys.
2. The format should be same as other agents